### PR TITLE
Minor Changes to the date dialogue

### LINKED ISSUE #5934: Two minor changes in the Use Date dialogue
@Ivanluv I hope these are both quick and easy to make.

1. The dialogue currently saves all the new variables at the end of the sheet.  Could this be changed to be just after the date variable.
2. The current order of the variables causes an unnecessary problem in the (later) define climatic data.  When the month is not January and two year variables are generated, then currently the factor variable is stored before the numeric variable.  Please could these be changed round, so the numeric variable is generated, or at least saved, before the factor variable for the date.

### PR DESCRIPTION
**Pull request by Ivanluv**
@africanmathsinitiative/developers  this fixes #5934

### CODE REVIEW

**Review by rdstern — APPROVED** [review]
@Ivanluv this is great.  Ready for @lloyddewit

**Comment by lloyddewit** [comment]
@shadrackkibet Please could you review? thanks

File: instat/static/InstatObject/R/data_object_R6.R [hunk 1]
```diff
@@ -2224,135 +2293,68 @@ DataSheet$set("public","split_date", function(col_name = "", year_val = FALSE, y
...
-	  month.list <- c("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec")
-	  month_levels <- if (s_start_month == 1) month.list else c(tail(month.list, -s_start_month + 1), head(month.list, s_start_month - 1))
-	  dekad_levels <- paste0(rep(month_levels, each = 3), 1:3)
-	  dekad_abbr_vector <- factor(paste(month_abbr_vector, dekad_val_vector, sep = ""), levels = dekad_levels)
-	  col_name <- next_default_item(prefix = "dekad_abbr", existing_names = self$get_column_names(), include_index = FALSE)
-    self$add_columns_to_data(col_name = col_name, col_data = dekad_abbr_vector)
-  }
-  if(pentad_val) {
-  	pentad_val_vector <- ((as.integer(pentad(col_data))) - (s_start_month - 1)*6) %% 72
-	  pentad_val_vector <- ifelse(pentad_val_vector == 0, 72, pentad_val_vector)
```
Comment (shadrackkibet) [inline instat/static/InstatObject/R/data_object_R6.R:2317]: I am struggling to know what exactly changed here. It is fuzzy. Apart from changing the order in which columns are added into the data frame and passing in `adjacent_column = org_name` anything else that changed?

Comment (Ivanluv) [reply instat/static/InstatObject/R/data_object_R6.R:2317]: No,theres nothing  else that changed

**Comment by lloyddewit** [comment]
@shadrackkibet please could you confirm that your review is complete? thanks

**Comment by shadrackkibet** [comment]
I am not confident that this is working as expected because I cannot trace what exactly changed. I would like someone to check. I did a test using Dodoma dataset and checked all checkboxes in the dialog. I got below error.
image
image

**Comment by lloyddewit** [comment]
@shadrackkibet I tried to do the same steps as you, but I wasn't able to recreate the error:

image

As far as I can see, my R command is the same as yours except that your `week_val` and `month_name` parameters are the opposite way around than I would expect. I don't know if this causes the bug but it probably should be corrected anyway.

**Comment by lloyddewit** [comment]
@Ivanluv please could you clarify the status of this PR? thanks

**Comment by lilyclements** [comment]
> I am not confident that this is working as expected because I cannot trace what exactly changed. I would like someone to check.

@shadrackkibet I thought I should check as well. I also did not get the same error. However, I did get an error upon reopening the dialog (when the reopen occurs after selecting everything and pressing OK):

image

**Comment by Ivanluv** [comment]
@lilyclements  I was not able to  reproduce the error you found

**Comment by lilyclements** [comment]
@Ivanluv interesting. I updated the branch from the master and am still getting this error. Can you try updating the branch as well and seeing if you then get the error? If you do not then perhaps we should skype.

**Comment by lilyclements** [comment]
@shadrackkibet if I use the RDS dodoma file I get the same error as you

image

If I then reopen the dialog, I get the `data_name` error I mentioned above. Do you get this error as well when you reopen?

**Comment by shadrackkibet** [comment]
Yes

**Comment by Ivanluv** [comment]
@lilyclements  I've realised that inserting any new  column just before the date column in `split_date` function  changes the climatic property   of the dataframe  from `TRUE`  to `""` thats  why this error is occuring
image . 
 I however dont know how to solve this.Could you help me with this

**Comment by Patowhiz** [comment]
@Ivanluv try merging the latest changes from the master. I made some changes to the `add_column` function.

I've looked at your code. I'm not sure about the huge changes you have made. Probably it's Github. But so far I can see you are trying to determine the position of the new columns every time you add relative to date column, I think that will change the order of the new columns too .

To address @rdstern 2 items in the issue, I would have adjusted the function script code to do the following;

1. Make the year numeric column be added before the year name factor column.(By just switching their current order of code execution).
2. Let the other columns be added in the current order. Then after all the columns are added, define a reordered vector character of the column names(based on the new proposed order) and call the reorder function. 

I find the above "algorithm" simplistic and easier to implement and will solve the 2 items mentioned. You may have a better way. I'm happy to discuss with you on the technical implementation if you find my above proposal to be easier.

**Comment by Ivanluv** [comment]
@Patowhiz could you look at this change

**Comment by Patowhiz** [comment]
@Ivanluv I can see you changed the order in which the columns are added after the selected "Date" column. This looks good.

**Comment by lloyddewit** [comment]
@shadrackkibet @lilyclements in Sep/Oct you were both able to trigger an error in this PR. @Ivanluv has since made some changes. Please could you confirm that this error is solved? thank you!

**Comment by lilyclements** [comment]
Can confirm for me that there is no longer an error in this PR

**Comment by shadrackkibet** [comment]
The error is no longer there.

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern the implementtion changed slightly since your last approval, please could you approve again? thanks

@dannyparsons `data_object_R6.R` was updated in this PR. The change was to change the order that the columns are added, and to add `adjacent_column = adjacent_column, before = FALSE` parameters to `self$add_columns_to_data(...`. The change looks low risk but if you'd like to review before merge then let me know.

**Review by rdstern — APPROVED** [review]
Looks fine

**Comment by dannyparsons** [comment]
@lloyddewit happy with those changes that you mention.

**Pull request merged**